src/test/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/UploadReportsStepTest.groovy

```diff
@@ -0,0 +1,308 @@
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl
+import de.tracetronic.jenkins.plugins.ecutestexecution.clients.RestApiClientFactory
+import de.tracetronic.jenkins.plugins.ecutestexecution.model.UploadResult
+import hudson.model.Item
+import hudson.security.ACL
+import hudson.util.FormValidation
+import jenkins.model.Jenkins
+import jenkins.security.MasterToSlaveCallable
+import org.junit.Rule
+import org.jvnet.hudson.test.JenkinsRule
+import spock.lang.Specification
+import hudson.EnvVars
+import hudson.Launcher
+import hudson.model.Run
+import hudson.model.Job
+import hudson.model.TaskListener
+import hudson.remoting.VirtualChannel
+import org.jenkinsci.plugins.workflow.steps.StepContext
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials
+import de.tracetronic.jenkins.plugins.ecutestexecution.clients.RestApiClient
+import de.tracetronic.jenkins.plugins.ecutestexecution.model.AdditionalSetting
+import spock.lang.Unroll
+
+class UploadReportsStepTest extends Specification {
+
+    @Rule
+    JenkinsRule jenkins = new JenkinsRule()
+
+    def stepContext
+    def launcher
+    def channel
+    def envVars
+    def listener
+    def run
+    def job
+    def apiClient
+    def credentials
+    static List<AdditionalSetting> additionalSettings = [
+        new AdditionalSetting("setting1", "value1"),
+        new AdditionalSetting("", "empty"),
+        new AdditionalSetting("setting2", "value2")
+    ]
+
+    void setup() {
+        stepContext = Mock(StepContext)
+        launcher = Mock(Launcher)
+        channel = Mock(VirtualChannel)
+        envVars = Mock(EnvVars)
+        listener = Mock(TaskListener)
+        run = Mock(Run)
+        job = Mock(Job)
+        apiClient = Mock(RestApiClient)
+        credentials = Mock(StandardUsernamePasswordCredentials)
+
+        launcher.getChannel() >> channel
+        stepContext.get(Launcher.class) >> launcher
+        stepContext.get(EnvVars.class) >> envVars
+        stepContext.get(TaskListener.class) >> listener
+        stepContext.get(Run.class) >> run
+        run.getParent() >> job
+   }
+
+    def "Constructor should initialize with default values"() {
+        when:
+            def step = new UploadReportsStep("http://localhost:8085", "auth")
+
+        then:
+            step.testGuideUrl == "http://localhost:8085"
+            step.credentialsId == "auth"
+            step.projectId == 1
+            step.useSettingsFromServer == true
+            step.additionalSettings == []
+            step.reportIds == []
+    }
+
+    def "getTestGuideUrl should trim trailing slash from testGuideUrl"() {
+        when:
+            def step = new UploadReportsStep("http://localhost:8085/", "auth")
+
+        then:
+            step.getTestGuideUrl() == "http://localhost:8085"
+    }
+
+    def "setAdditionalSettings should '#scenario'"() {
```

Comment:
instead of using a scenario name the test method should named clearly with dynamic properties